RELEASES.md

```diff
@@ -232,6 +237,22 @@ completed, open a PR using the process above.
 Only when the bug is not seen in main and must be made for the specific
 release branch should you open a PR with new code.
 
+### Upgrade Path
+
+The upgrade path for containerd is such that the 0.0.x patch releases are
+always backward compatible with its major and minor version. Minor (0.x.0)
+version will always be compatible with the previous minor release. i.e. 1.2.0
+is backwards compatible with 1.1.0 and 1.1.0 is compatible with 1.0.0. There is
+no compatibility guarantees for upgrades that span multiple, _minor_ releases.
+For example, 1.0.0 to 1.2.0 is not supported. One should first upgrade to 1.1,
+then 1.2.
+
+There are no compatibility guarantees with upgrades to _major_ versions. For 2.0, migration was
+added to ensure upgrading from 1.6 or 1.7 to 2.0 is easy. The latest releases of 1.6 and 1.7 provide
+deprecation warnings if any configuration is used which is incompatible with 2.0. If deprecation
+warnings are showing up, the configuration can be safely upgrade in 1.6 or 1.7 before upgrading to
+2.0. Once no deprecation warnings are showing up, the upgrade to 2.0 is completely compatible.
```

Comment:
"completely compatible" is a pretty strong statement.  There are some things that we can't detect in the deprecation warning system and thus can't produce warnings for.  Examples for 2.0 include the `io_uring_*` seccomp change and the `LimitNOFILE` change in the systemd unit.  Major version bumps are allowed to include breaking changes, and the deprecation warning system is a best-effort system (though I have tried to make it as accurate as possible; it shouldn't have false-positives but can have false-negatives).  Our advice here should recommend reviewing the release notes for breaking changes and the table in this document.

Reply:
Updated